lib/data/locationConstraintParser.js

```diff
@@ -41,26 +42,39 @@ function parseLC() {
             clients[location].clientType = 'scality';
         }
         if (locationObj.type === 'aws_s3' || locationObj.type === 'gcp') {
-            if (process.env.CI_PROXY === 'true') {
-                locationObj.details.https = false;
-                locationObj.details.proxy = proxyAddress;
+            let connectionAgent;
+            const endpoint = locationObj.type === 'gcp' ?
+                locationObj.details.gcpEndpoint :
+                locationObj.details.awsEndpoint;
+            const proxyMatch = proxyCompareURL(endpoint);
+            if (config.outboundProxy.url && !proxyMatch) {
+                const options = url.parse(config.outboundProxy.url);
+                if (config.outboundProxy.certFilePaths) {
+                    const { keypath, certpath, capath } =
+                        config.outboundProxy.certFilePaths;
+                    options.cert = certpath ?
+                        fs.readFileSync(certpath, 'ascii') : undefined;
```

Comment:
Same thing, if you read the certs into an object in Config, you don't have to redo this in multiple places.